samcli/lib/test_runner/test_runner_template_generator.py

```diff
@@ -0,0 +1,347 @@
+import json
+import boto3
+import re
+import logging
+from jinja2 import Template
+from cfn_flip import to_yaml
+from typing import *
+
+LOG = logging.getLogger(__name__)
+
+
+def __extract_api_id_from_arn(api_arn: str) -> str:
+    """
+    Extracts an api id from an HTTP or REST api arn.
+
+    NOTE: https://docs.aws.amazon.com/apigateway/latest/developerguide/arn-format-reference.html
+
+    REST API Arn: `arn:partition:apigateway:region::/restapis/api-id`
+
+    HTTP API Arn: `arn:partition:apigateway:region::/apis/api-id`
+
+    Parameters
+    ----------
+    api_arn : str
+        The arn of the HTTP or REST api from which the api id is extracted.
+
+    Returns
+    -------
+    str
+        The api id from the api arn.
+    """
+
+    return api_arn[api_arn.rindex("/") + 1 :]
+
+
+def __get_permissions_map() -> dict:
+    """
+    Returns a dictionary mapping a Regular Expression representing the arn of a specific resource to the default permissions generated for that resource.
+
+    Returns
+    -------
+    default_permissions_map : dict
+        A dictionary mapping a Regular Expression representing the arn of a resource to a list of IAM Action permissions generated for that resource.
+    """
+
+    partition_ex = "(aws|aws-cn|aws-us-gov)"
+    region_ex = "\w+-\w+-\d"
```

Comment:
This won't work for GovCloud regions.

Reply:
Replacing with `(us(-gov)?|ap|ca|cn|eu|sa)-(central|(north|south)?(east|west)?)-\d`